gtsam/hybrid/DiscreteBoundaryFactor.cpp

```diff
@@ -0,0 +1,104 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ *  @file DiscreteBoundaryFactor.cpp
+ *  @date Mach, 2025
+ *  @author Varun Agrawal
+ */
+
+#include <gtsam/hybrid/DiscreteBoundaryFactor.h>
```

Comment:
Again, not sure if you guys use the Google style guide, but if so, consider [including what you use](https://google.github.io/styleguide/cppguide.html#Include_What_You_Use) rather than relying on transitive includes here.